docs/contributor/adr/0007-compatibilityMode.md

```diff
@@ -0,0 +1,52 @@
+# Support for **compatibilityMode** Configuration
+
+## Status
+
+Accepted by the Istio module team
+
+## Context
+
+With Istio's introduction of a [compatibilityVersion](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/#easing-upgrades-with-compatibility-versions) and environment variables allowing to mitigate breaking changes during Istio upgrade, Istio module team decided to introduce compatibility mode to allow users leverage possibility of remaining compatible with previous Istio versions.
+
+## Considerations
```

Comment:
Consideration is not part of our ADR template 🤔 This is just part of the decision, but you might need to write it different. I would not write it as a question, but mention what were the options, what was the decision and why the decision was made.

Reply:
Then I'd remove this part, because we have other part that explains decision